src/seq/ChainWrapper.js

```diff
@@ -49,16 +51,51 @@ class ChainWrapper {
    * @param {function} fn The function to call.
    * @param {Array|string} path The path of the property to be set.
    * @param {...*} args The arguments for the function call.
-   * @returns {seq.ChainWrapper} The wrapper instance.
+   * @returns {seq.ChainWrapper} The new wrapper instance.
    * @since 0.1.8
    */
   _call(fn, path, args) {
-    this._flow.push(object => fn(object, this._absolutePath(path), ...args))
-    return this
+    return new ChainWrapper(
+      this._wrapped,
+      this._path,
+      concat(this._flow, object => fn(object, this._absolutePath(path), ...args)),
+    )
   }
 
   /**
-   * Executes the chain sequence to resolve the unwrapped object.
+   * Executes the chain sequence.
+   * @returns {seq.ChainWrapper} The new wrapper instance.
+   * @since 0.3.0
+   */
+  commit() {
+    if (this._flow.length === 0) return this
+
+    if (this._commited === null)
+      this._commited = new ChainWrapper(
+        flow(this._flow)(this._wrapped),
+        this._path,
+      )
+
+    return this._commited
+  }
+
+  /**
+   * Executes the chain sequence and calls <code>callback</code> with the unwrapped object.
+   * @param {seq.peekCallback} callback Function to be called with the resolved unwrapped value.
+   * @returns {seq.ChainWrapper} The new wrapper instance.
+   * @todo Add an example.
+   * @since 0.3.0
+   */
+  peek(callback) {
```

Comment:
@jlandure yes it is the same.
The name `peek` is taken from java 8 [Stream api](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#peek-java.util.function.Consumer-).
`tap` means "listen secretly", as when you tap a phone line.
I think `peek` meaning "having a quick look at" is a more appropriate name.